lib/monitoring/middleware/prometheus_collector.rb

```diff
@@ -0,0 +1,52 @@
+require 'benchmark'
+
+module Monitoring
+  module Middleware
+    class PrometheusCollector
+      attr_reader :app
+
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env) # :nodoc:
+        trace(env) { @app.call(env) }
+      end
+
+      protected
+
+      # Trace HTTP requests
+      def trace(env)
+        response = nil
+        duration = Benchmark.realtime { response = yield }
+        record(env, response.first.to_s, duration)
+        return response
+      rescue => exception
+        Monitoring::PubSub.instance.publish(
+          "conjur.request_exception", 
+          exception: exception
+        )
+        raise
+      end
+
+      def record(env, code, duration)
+        Monitoring::PubSub.instance.publish(
+          "conjur.request", 
+          code: code,
+          method: env['REQUEST_METHOD'],
+          path: strip_ids_from_path(env['PATH_INFO']),
```

Comment:
In the solution design we landed on having the labels `code` and the open-spec `operation` identifier, see https://github.com/cyberark/conjur/blob/metrics-solution-design/design/telemetry/metrics_solution_design.md#conjur-http-requests. There's a reference implementation from the hackathon at  https://github.com/cyberark/conjur/blob/bdfc77a03cefd1e6667a56c10111bbefc23b61ac/lib/monitoring/metrics/endpoints.rb#L64.